MacOS tests raise `AttributeError` in asyncio

### Describe the problem.

I have observed twice now a MacOS test job failing due to an `AttributeError` when awaiting a cancelled task from aiortc. The issue seems to be from within asyncio. 

I found a similar issue to this from another asyncio library: https://github.com/Tribler/tribler/issues/5830

```
2023-10-30T19:11:01.6713507Z Requested labels: macos-latest
2023-10-30T19:11:01.6713848Z Job defined at: proxystore/proxystore/.github/workflows/tests.yml@refs/pull/404/merge
2023-10-30T19:11:01.6713968Z Waiting for a runner to pick up this job...
2023-10-30T19:11:03.0204161Z Job is waiting for a hosted runner to come online.
2023-10-30T19:11:06.6723986Z Job is about to start running on the hosted runner: GitHub Actions 19 (hosted)
2023-10-30T19:11:10.5193640Z Current runner version: '2.311.0'
2023-10-30T19:11:10.5224770Z ##[group]Operating System
2023-10-30T19:11:10.5225710Z macOS
2023-10-30T19:11:10.5226090Z 12.6.9
2023-10-30T19:11:10.5226460Z 21G726
2023-10-30T19:11:10.5226840Z ##[endgroup]
2023-10-30T19:11:10.5227280Z ##[group]Runner Image
2023-10-30T19:11:10.5227820Z Image: macos-12
2023-10-30T19:11:10.5228260Z Version: 20230921.1
2023-10-30T19:11:10.5229680Z Included Software: https://github.com/actions/runner-images/blob/macOS-12/20230921.1/images/macos/macos-12-Readme.md
2023-10-30T19:11:10.5231940Z Image Release: https://github.com/actions/runner-images/releases/tag/macOS-12%2F20230921.1
2023-10-30T19:11:10.5233250Z ##[endgroup]
2023-10-30T19:11:10.5233730Z ##[group]Runner Image Provisioner
2023-10-30T19:11:10.5234370Z 2.0.312.1
2023-10-30T19:11:10.5234750Z ##[endgroup]
2023-10-30T19:11:10.5238260Z ##[group]GITHUB_TOKEN Permissions
2023-10-30T19:11:10.5240290Z Actions: write
2023-10-30T19:11:10.5240760Z Checks: write
2023-10-30T19:11:10.5241360Z Contents: write
2023-10-30T19:11:10.5241860Z Deployments: write
2023-10-30T19:11:10.5242360Z Discussions: write
2023-10-30T19:11:10.5242830Z Issues: write
2023-10-30T19:11:10.5243270Z Metadata: read
2023-10-30T19:11:10.5243720Z Packages: write
2023-10-30T19:11:10.5244170Z Pages: write
2023-10-30T19:11:10.5244720Z PullRequests: write
2023-10-30T19:11:10.5245260Z RepositoryProjects: write
2023-10-30T19:11:10.5245840Z SecurityEvents: write
2023-10-30T19:11:10.5246350Z Statuses: write
2023-10-30T19:11:10.5246810Z ##[endgroup]
2023-10-30T19:11:10.5250890Z Secret source: Actions
2023-10-30T19:11:10.5251590Z Prepare workflow directory
2023-10-30T19:11:10.6208270Z Prepare all required actions
2023-10-30T19:11:10.6434940Z Getting action download info
2023-10-30T19:11:10.8838610Z Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
2023-10-30T19:11:12.0058980Z Download action repository 'actions/setup-python@v4' (SHA:65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236)
2023-10-30T19:11:12.8455710Z Download action repository 'actions/cache@v3' (SHA:704facf57e6136b1bc63b828d79edcd491f0ee84)
2023-10-30T19:11:13.7842220Z Complete job name: tests (macos-latest, 3.9, py310)
2023-10-30T19:11:13.8912730Z ##[group]Run actions/checkout@v4
2023-10-30T19:11:13.8913400Z with:
2023-10-30T19:11:13.8913830Z   repository: proxystore/proxystore
2023-10-30T19:11:13.8914760Z   token: ***
2023-10-30T19:11:13.8915180Z   ssh-strict: true
2023-10-30T19:11:13.8915640Z   persist-credentials: true
2023-10-30T19:11:13.8916150Z   clean: true
2023-10-30T19:11:13.8916600Z   sparse-checkout-cone-mode: true
2023-10-30T19:11:13.8917190Z   fetch-depth: 1
2023-10-30T19:11:13.8917610Z   fetch-tags: false
2023-10-30T19:11:13.8918070Z   show-progress: true
2023-10-30T19:11:13.8918520Z   lfs: false
2023-10-30T19:11:13.8918930Z   submodules: false
2023-10-30T19:11:13.8919400Z   set-safe-directory: true
2023-10-30T19:11:13.8919900Z ##[endgroup]
2023-10-30T19:11:14.3406810Z Syncing repository: proxystore/proxystore
2023-10-30T19:11:14.3410150Z ##[group]Getting Git version info
2023-10-30T19:11:14.3411490Z Working directory is '/Users/runner/work/proxystore/proxystore'
2023-10-30T19:11:14.3413010Z [command]/usr/local/bin/git version
2023-10-30T19:11:14.3581590Z git version 2.42.0
2023-10-30T19:11:14.3636530Z ##[endgroup]
2023-10-30T19:11:14.3657780Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/52e8809a-5b31-4974-b4c8-c8d9d51516d6/.gitconfig'
2023-10-30T19:11:14.3678690Z Temporarily overriding HOME='/Users/runner/work/_temp/52e8809a-5b31-4974-b4c8-c8d9d51516d6' before making global git config changes
2023-10-30T19:11:14.3681520Z Adding repository directory to the temporary git global config as a safe directory
2023-10-30T19:11:14.3684740Z [command]/usr/local/bin/git config --global --add safe.directory /Users/runner/work/proxystore/proxystore
2023-10-30T19:11:14.3787600Z Deleting the contents of '/Users/runner/work/proxystore/proxystore'
2023-10-30T19:11:14.3794840Z ##[group]Initializing the repository
2023-10-30T19:11:14.3800340Z [command]/usr/local/bin/git init /Users/runner/work/proxystore/proxystore
2023-10-30T19:11:14.4017420Z hint: Using 'master' as the name for the initial branch. This default branch name
2023-10-30T19:11:14.4020180Z hint: is subject to change. To configure the initial branch name to use in all
2023-10-30T19:11:14.4021680Z hint: of your new repositories, which will suppress this warning, call:
2023-10-30T19:11:14.4022990Z hint: 
2023-10-30T19:11:14.4023780Z hint: 	git config --global init.defaultBranch <name>
2023-10-30T19:11:14.4024580Z hint: 
2023-10-30T19:11:14.4025770Z hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
2023-10-30T19:11:14.4027240Z hint: 'development'. The just-created branch can be renamed via this command:
2023-10-30T19:11:14.4028320Z hint: 
2023-10-30T19:11:14.4029180Z hint: 	git branch -m <name>
2023-10-30T19:11:14.4051570Z Initialized empty Git repository in /Users/runner/work/proxystore/proxystore/.git/
2023-10-30T19:11:14.4064960Z [command]/usr/local/bin/git remote add origin https://github.com/proxystore/proxystore
2023-10-30T19:11:14.4159960Z ##[endgroup]
2023-10-30T19:11:14.4160930Z ##[group]Disabling automatic garbage collection
2023-10-30T19:11:14.4266500Z [command]/usr/local/bin/git config --local gc.auto 0
2023-10-30T19:11:14.4269640Z ##[endgroup]
2023-10-30T19:11:14.4270500Z ##[group]Setting up auth
2023-10-30T19:11:14.4271700Z [command]/usr/local/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-10-30T19:11:14.4336290Z [command]/usr/local/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-10-30T19:11:14.5607390Z [command]/usr/local/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-10-30T19:11:14.5685870Z [command]/usr/local/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-10-30T19:11:14.7632530Z [command]/usr/local/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2023-10-30T19:11:14.7650360Z ##[endgroup]
2023-10-30T19:11:14.7656270Z ##[group]Fetching the repository
2023-10-30T19:11:14.7669470Z [command]/usr/local/bin/git -c protocol.version=2 fetch --no-tags --prune --no-recurse-submodules --depth=1 origin +8998847b7e4f385004fe71940afc909454a0def6:refs/remotes/pull/404/merge
2023-10-30T19:11:15.4074010Z From https://github.com/proxystore/proxystore
2023-10-30T19:11:15.4076960Z  * [new ref]         8998847b7e4f385004fe71940afc909454a0def6 -> pull/404/merge
2023-10-30T19:11:15.4172120Z ##[endgroup]
2023-10-30T19:11:15.4174210Z ##[group]Determining the checkout info
2023-10-30T19:11:15.4175870Z ##[endgroup]
2023-10-30T19:11:15.4177350Z ##[group]Checking out the ref
2023-10-30T19:11:15.4180260Z [command]/usr/local/bin/git checkout --progress --force refs/remotes/pull/404/merge
2023-10-30T19:11:15.5085770Z HEAD is now at 8998847 Merge 490b80b8f3f1a249c08081aa37b8777f01d2129d into 0aa01f96c9b1292b27873f8b0bc23e891666fc1a
2023-10-30T19:11:15.5094810Z ##[endgroup]
2023-10-30T19:11:15.5237750Z [command]/usr/local/bin/git log -1 --format='%H'
2023-10-30T19:11:15.5340430Z '8998847b7e4f385004fe71940afc909454a0def6'
2023-10-30T19:11:15.5906160Z ##[group]Run actions/setup-python@v4
2023-10-30T19:11:15.5906870Z with:
2023-10-30T19:11:15.5907310Z   python-version: 3.9
2023-10-30T19:11:15.5907840Z   check-latest: false
2023-10-30T19:11:15.5908680Z   token: ***
2023-10-30T19:11:15.5909180Z   update-environment: true
2023-10-30T19:11:15.5909780Z   allow-prereleases: false
2023-10-30T19:11:15.5910320Z ##[endgroup]
2023-10-30T19:11:16.2832610Z ##[group]Installed versions
2023-10-30T19:11:16.2968160Z Successfully set up CPython (3.9.18)
2023-10-30T19:11:16.2970890Z ##[endgroup]
2023-10-30T19:11:16.3219520Z ##[group]Run echo "PIP_CACHE_DIR=$(pip cache dir)" >> $GITHUB_ENV
2023-10-30T19:11:16.3221350Z [36;1mecho "PIP_CACHE_DIR=$(pip cache dir)" >> $GITHUB_ENV[0m
2023-10-30T19:11:16.3453440Z shell: /bin/bash -e {0}
2023-10-30T19:11:16.3454040Z env:
2023-10-30T19:11:16.3454750Z   pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.3456050Z   PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
2023-10-30T19:11:16.3457320Z   Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.3458430Z   Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.3459610Z   Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.3460510Z ##[endgroup]
2023-10-30T19:11:16.9176090Z ##[group]Run actions/cache@v3
2023-10-30T19:11:16.9176830Z with:
2023-10-30T19:11:16.9177410Z   path: /Users/runner/Library/Caches/pip
2023-10-30T19:11:16.9178710Z   key: macos-latest-pip-3.9-00b6a072d72a9a09c0b07311a82d84372f64b33805efcf6eed12e195fbe2c7b2
2023-10-30T19:11:16.9180050Z   restore-keys: macos-latest-pip-3.9-

2023-10-30T19:11:16.9181060Z   enableCrossOsArchive: false
2023-10-30T19:11:16.9181780Z   fail-on-cache-miss: false
2023-10-30T19:11:16.9182410Z   lookup-only: false
2023-10-30T19:11:16.9183480Z env:
2023-10-30T19:11:16.9184260Z   pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.9185590Z   PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
2023-10-30T19:11:16.9187040Z   Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.9188730Z   Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.9190460Z   Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:16.9191540Z   PIP_CACHE_DIR: /Users/runner/Library/Caches/pip
2023-10-30T19:11:16.9192310Z ##[endgroup]
2023-10-30T19:11:18.8410170Z Received 0 of 73974249 (0.0%), 0.0 MBs/sec
2023-10-30T19:11:19.8440200Z Received 16777216 of 73974249 (22.7%), 8.0 MBs/sec
2023-10-30T19:11:20.8447730Z Received 20971520 of 73974249 (28.3%), 6.7 MBs/sec
2023-10-30T19:11:21.8673860Z Received 37748736 of 73974249 (51.0%), 9.0 MBs/sec
2023-10-30T19:11:22.8443420Z Received 53002729 of 73974249 (71.7%), 10.1 MBs/sec
2023-10-30T19:11:23.8446110Z Received 65585641 of 73974249 (88.7%), 10.4 MBs/sec
2023-10-30T19:11:24.8612760Z Received 65585641 of 73974249 (88.7%), 8.9 MBs/sec
2023-10-30T19:11:25.2563190Z Cache Size: ~71 MB (73974249 B)
2023-10-30T19:11:25.3712360Z [command]/usr/local/bin/gtar -xf /Users/runner/work/_temp/bff6cdc9-81e3-415e-91b3-24dfba0a81db/cache.tzst -P -C /Users/runner/work/proxystore/proxystore --delay-directory-restore --use-compress-program unzstd
2023-10-30T19:11:25.8528050Z Received 73974249 of 73974249 (100.0%), 8.8 MBs/sec
2023-10-30T19:11:26.0367470Z Cache restored successfully
2023-10-30T19:11:26.0755890Z Cache restored from key: macos-latest-pip-3.9-00b6a072d72a9a09c0b07311a82d84372f64b33805efcf6eed12e195fbe2c7b2
2023-10-30T19:11:26.1095610Z ##[group]Run pip install --upgrade setuptools pip tox virtualenv
2023-10-30T19:11:26.1097680Z [36;1mpip install --upgrade setuptools pip tox virtualenv[0m
2023-10-30T19:11:26.1392030Z shell: /bin/bash -e {0}
2023-10-30T19:11:26.1392590Z env:
2023-10-30T19:11:26.1393330Z   pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:26.1394700Z   PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
2023-10-30T19:11:26.1396060Z   Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:26.1397230Z   Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:26.1398430Z   Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:26.1399500Z   PIP_CACHE_DIR: /Users/runner/Library/Caches/pip
2023-10-30T19:11:26.1400300Z ##[endgroup]
2023-10-30T19:11:27.8078200Z Requirement already satisfied: setuptools in /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages (58.1.0)
2023-10-30T19:11:28.3198390Z Collecting setuptools
2023-10-30T19:11:28.3211470Z   Obtaining dependency information for setuptools from https://files.pythonhosted.org/packages/bb/26/7945080113158354380a12ce26873dd6c1ebd88d47f5bc24e2c5bb38c16a/setuptools-68.2.2-py3-none-any.whl.metadata
2023-10-30T19:11:28.3345570Z   Using cached setuptools-68.2.2-py3-none-any.whl.metadata (6.3 kB)
2023-10-30T19:11:28.3392150Z Requirement already satisfied: pip in /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages (23.0.1)
2023-10-30T19:11:28.5479020Z Collecting pip
2023-10-30T19:11:28.5483700Z   Obtaining dependency information for pip from https://files.pythonhosted.org/packages/47/6a/453160888fab7c6a432a6e25f8afe6256d0d9f2cbd25971021da6491d899/pip-23.3.1-py3-none-any.whl.metadata
2023-10-30T19:11:28.5730690Z   Downloading pip-23.3.1-py3-none-any.whl.metadata (3.5 kB)
2023-10-30T19:11:28.7566500Z Collecting tox
2023-10-30T19:11:28.7572280Z   Obtaining dependency information for tox from https://files.pythonhosted.org/packages/f5/f9/963052e8b825645c54262dce7b7c88691505e3b9ee10a3e3667711eaaf21/tox-4.11.3-py3-none-any.whl.metadata
2023-10-30T19:11:28.7614780Z   Using cached tox-4.11.3-py3-none-any.whl.metadata (5.0 kB)
2023-10-30T19:11:28.9391830Z Collecting virtualenv
2023-10-30T19:11:28.9549960Z   Obtaining dependency information for virtualenv from https://files.pythonhosted.org/packages/7f/19/1f0eddcb9acf00a95793ce83417f69e0fd106c192121360af499cd6fde39/virtualenv-20.24.6-py3-none-any.whl.metadata
2023-10-30T19:11:28.9559510Z   Downloading virtualenv-20.24.6-py3-none-any.whl.metadata (4.5 kB)
2023-10-30T19:11:29.1017080Z Collecting cachetools>=5.3.1 (from tox)
2023-10-30T19:11:29.1020320Z   Obtaining dependency information for cachetools>=5.3.1 from https://files.pythonhosted.org/packages/a2/91/2d843adb9fbd911e0da45fbf6f18ca89d07a087c3daa23e955584f90ebf4/cachetools-5.3.2-py3-none-any.whl.metadata
2023-10-30T19:11:29.1099650Z   Downloading cachetools-5.3.2-py3-none-any.whl.metadata (5.2 kB)
2023-10-30T19:11:29.1413320Z Collecting chardet>=5.2 (from tox)
2023-10-30T19:11:29.1418770Z   Obtaining dependency information for chardet>=5.2 from https://files.pythonhosted.org/packages/38/6f/f5fbc992a329ee4e0f288c1fe0e2ad9485ed064cac731ed2fe47dcc38cbf/chardet-5.2.0-py3-none-any.whl.metadata
2023-10-30T19:11:29.1464290Z   Using cached chardet-5.2.0-py3-none-any.whl.metadata (3.4 kB)
2023-10-30T19:11:29.1823620Z Collecting colorama>=0.4.6 (from tox)
2023-10-30T19:11:29.1867980Z   Using cached colorama-0.4.6-py2.py3-none-any.whl (25 kB)
2023-10-30T19:11:29.2427870Z Collecting filelock>=3.12.3 (from tox)
2023-10-30T19:11:29.2432140Z   Obtaining dependency information for filelock>=3.12.3 from https://files.pythonhosted.org/packages/81/54/84d42a0bee35edba99dee7b59a8d4970eccdd44b99fe728ed912106fc781/filelock-3.13.1-py3-none-any.whl.metadata
2023-10-30T19:11:29.2505590Z   Downloading filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)
2023-10-30T19:11:29.3048070Z Collecting packaging>=23.1 (from tox)
2023-10-30T19:11:29.3053380Z   Obtaining dependency information for packaging>=23.1 from https://files.pythonhosted.org/packages/ec/1a/610693ac4ee14fcdf2d9bf3c493370e4f2ef7ae2e19217d7a237ff42367d/packaging-23.2-py3-none-any.whl.metadata
2023-10-30T19:11:29.3102080Z   Using cached packaging-23.2-py3-none-any.whl.metadata (3.2 kB)
2023-10-30T19:11:29.3534650Z Collecting platformdirs>=3.10 (from tox)
2023-10-30T19:11:29.3539810Z   Obtaining dependency information for platformdirs>=3.10 from https://files.pythonhosted.org/packages/56/29/3ec311dc18804409ecf0d2b09caa976f3ae6215559306b5b530004e11156/platformdirs-3.11.0-py3-none-any.whl.metadata
2023-10-30T19:11:29.3581020Z   Using cached platformdirs-3.11.0-py3-none-any.whl.metadata (11 kB)
2023-10-30T19:11:29.4093580Z Collecting pluggy>=1.3 (from tox)
2023-10-30T19:11:29.4097750Z   Obtaining dependency information for pluggy>=1.3 from https://files.pythonhosted.org/packages/05/b8/42ed91898d4784546c5f06c60506400548db3f7a4b3fb441cba4e5c17952/pluggy-1.3.0-py3-none-any.whl.metadata
2023-10-30T19:11:29.4143770Z   Using cached pluggy-1.3.0-py3-none-any.whl.metadata (4.3 kB)
2023-10-30T19:11:29.4411380Z Collecting pyproject-api>=1.6.1 (from tox)
2023-10-30T19:11:29.4415900Z   Obtaining dependency information for pyproject-api>=1.6.1 from https://files.pythonhosted.org/packages/cf/b4/39eea50542e50e93876ebc09c4349a9c9eee9f6b9c9d30f88c7dc5433db8/pyproject_api-1.6.1-py3-none-any.whl.metadata
2023-10-30T19:11:29.4456890Z   Using cached pyproject_api-1.6.1-py3-none-any.whl.metadata (2.8 kB)
2023-10-30T19:11:29.4772920Z Collecting tomli>=2.0.1 (from tox)
2023-10-30T19:11:29.4816820Z   Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
2023-10-30T19:11:29.5553030Z Collecting distlib<1,>=0.3.7 (from virtualenv)
2023-10-30T19:11:29.5557790Z   Obtaining dependency information for distlib<1,>=0.3.7 from https://files.pythonhosted.org/packages/43/a0/9ba967fdbd55293bacfc1507f58e316f740a3b231fc00e3d86dc39bc185a/distlib-0.3.7-py2.py3-none-any.whl.metadata
2023-10-30T19:11:29.5600690Z   Using cached distlib-0.3.7-py2.py3-none-any.whl.metadata (5.1 kB)
2023-10-30T19:11:29.6546120Z Using cached setuptools-68.2.2-py3-none-any.whl (807 kB)
2023-10-30T19:11:29.6971510Z Downloading pip-23.3.1-py3-none-any.whl (2.1 MB)
2023-10-30T19:11:29.7802920Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 26.5 MB/s eta 0:00:00
2023-10-30T19:11:29.7859900Z Using cached tox-4.11.3-py3-none-any.whl (153 kB)
2023-10-30T19:11:29.7984770Z Downloading virtualenv-20.24.6-py3-none-any.whl (3.8 MB)
2023-10-30T19:11:29.9335650Z    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 28.7 MB/s eta 0:00:00
2023-10-30T19:11:29.9424700Z Downloading cachetools-5.3.2-py3-none-any.whl (9.3 kB)
2023-10-30T19:11:29.9600520Z Using cached chardet-5.2.0-py3-none-any.whl (199 kB)
2023-10-30T19:11:29.9669480Z Using cached distlib-0.3.7-py2.py3-none-any.whl (468 kB)
2023-10-30T19:11:29.9766540Z Downloading filelock-3.13.1-py3-none-any.whl (11 kB)
2023-10-30T19:11:29.9933670Z Using cached packaging-23.2-py3-none-any.whl (53 kB)
2023-10-30T19:11:30.0003770Z Using cached platformdirs-3.11.0-py3-none-any.whl (17 kB)
2023-10-30T19:11:30.0082440Z Using cached pluggy-1.3.0-py3-none-any.whl (18 kB)
2023-10-30T19:11:30.0156250Z Using cached pyproject_api-1.6.1-py3-none-any.whl (12 kB)
2023-10-30T19:11:30.3800700Z Installing collected packages: distlib, tomli, setuptools, pluggy, platformdirs, pip, packaging, filelock, colorama, chardet, cachetools, virtualenv, pyproject-api, tox
2023-10-30T19:11:30.4907310Z   Attempting uninstall: setuptools
2023-10-30T19:11:30.4918450Z     Found existing installation: setuptools 58.1.0
2023-10-30T19:11:30.5170240Z     Uninstalling setuptools-58.1.0:
2023-10-30T19:11:30.5359020Z       Successfully uninstalled setuptools-58.1.0
2023-10-30T19:11:31.2210660Z   Attempting uninstall: pip
2023-10-30T19:11:31.2218360Z     Found existing installation: pip 23.0.1
2023-10-30T19:11:31.3072900Z     Uninstalling pip-23.0.1:
2023-10-30T19:11:31.4431590Z       Successfully uninstalled pip-23.0.1
2023-10-30T19:11:33.7027240Z Successfully installed cachetools-5.3.2 chardet-5.2.0 colorama-0.4.6 distlib-0.3.7 filelock-3.13.1 packaging-23.2 pip-23.2.1 platformdirs-3.11.0 pluggy-1.3.0 pyproject-api-1.6.1 setuptools-68.2.2 tomli-2.0.1 tox-4.11.3 virtualenv-20.24.6
2023-10-30T19:11:34.0540390Z ##[group]Run tox -e py310
2023-10-30T19:11:34.0541400Z [36;1mtox -e py310[0m
2023-10-30T19:11:34.0705190Z shell: /bin/bash -e {0}
2023-10-30T19:11:34.0705930Z env:
2023-10-30T19:11:34.0706650Z   pythonLocation: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:34.0707960Z   PKG_CONFIG_PATH: /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/pkgconfig
2023-10-30T19:11:34.0709420Z   Python_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:34.0710620Z   Python2_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:34.0711760Z   Python3_ROOT_DIR: /Users/runner/hostedtoolcache/Python/3.9.18/x64
2023-10-30T19:11:34.0712930Z   PIP_CACHE_DIR: /Users/runner/Library/Caches/pip
2023-10-30T19:11:34.0713690Z ##[endgroup]
2023-10-30T19:11:37.3013550Z .pkg: install_requires> python -I -m pip install 'setuptools>=64.0' setuptools_scm
2023-10-30T19:11:39.5863730Z .pkg: _optional_hooks> python /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta
2023-10-30T19:11:39.9027050Z .pkg: get_requires_for_build_sdist> python /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta
2023-10-30T19:11:40.4720300Z .pkg: freeze> python -m pip freeze --all
2023-10-30T19:11:40.9310920Z .pkg: packaging==23.2,pip==23.3.1,setuptools==68.2.2,setuptools-scm==8.0.4,tomli==2.0.1,typing_extensions==4.8.0,wheel==0.41.2
2023-10-30T19:11:40.9319510Z .pkg: build_sdist> python /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta
2023-10-30T19:11:41.4670770Z py310: install_package_deps> python -I -m pip install 'aiortc>=1.3.2' aiosqlite black 'click!=8.1.4' 'cloudpickle>=1.6.0' 'covdefaults>=2.2' coverage 'cryptography>=39.0.1' 'globus-sdk>=3.3.0' 'lazy-object-proxy>=1.6.0' mypy pre-commit psutil pytest pytest-asyncio pytest-cov pytest-timeout python-daemon 'quart>=0.18.0' 'redis>=3.4' 'requests>=2.27.1' ruff tosholi tox types-redis types-requests 'typing-extensions>=4.3.0' 'uvicorn[standard]' virtualenv 'websockets>=10.0'
2023-10-30T19:12:11.1081200Z py310: install_package> python -I -m pip install --force-reinstall --no-deps /Users/runner/work/proxystore/proxystore/.tox/.tmp/package/1/proxystore-0.6.1.dev1.tar.gz
2023-10-30T19:12:18.8208150Z py310: freeze> python -m pip freeze --all
2023-10-30T19:12:19.4954880Z py310: aiofiles==23.2.1,aioice==0.9.0,aiortc==1.5.0,aiosqlite==0.19.0,anyio==4.0.0,async-timeout==4.0.3,av==10.0.0,black==23.10.1,blinker==1.6.3,cachetools==5.3.2,certifi==2023.7.22,cffi==1.16.0,cfgv==3.4.0,chardet==5.2.0,charset-normalizer==3.3.1,click==8.1.7,cloudpickle==3.0.0,colorama==0.4.6,covdefaults==2.3.0,coverage==7.3.2,cryptography==41.0.5,dacite==1.8.1,distlib==0.3.7,dnspython==2.4.2,docutils==0.20.1,exceptiongroup==1.1.3,filelock==3.13.1,Flask==3.0.0,globus-sdk==3.30.0,google-crc32c==1.5.0,h11==0.14.0,h2==4.1.0,hpack==4.0.0,httptools==0.6.1,Hypercorn==0.15.0,hyperframe==6.0.1,identify==2.5.31,idna==3.4,ifaddr==0.2.0,iniconfig==2.0.0,itsdangerous==2.1.2,Jinja2==3.1.2,lazy-object-proxy==1.9.0,lockfile==0.12.2,MarkupSafe==2.1.3,mypy==1.6.1,mypy-extensions==1.0.0,nodeenv==1.8.0,packaging==23.2,pathspec==0.11.2,pip==23.3.1,platformdirs==3.11.0,pluggy==1.3.0,pre-commit==3.5.0,priority==2.0.0,proxystore @ file:///Users/runner/work/proxystore/proxystore/.tox/.tmp/package/1/proxystore-0.6.1.dev1.tar.gz#sha256=645d5ac3c6c4069691f09e8a69944584d84e66c19e207c27750e275644821b97,psutil==5.9.6,pycparser==2.21,pyee==11.0.1,PyJWT==2.8.0,pylibsrtp==0.8.0,pyOpenSSL==23.3.0,pyproject-api==1.6.1,pytest==7.4.3,pytest-asyncio==0.21.1,pytest-cov==4.1.0,pytest-timeout==2.2.0,python-daemon==3.0.1,python-dotenv==1.0.0,PyYAML==6.0.1,Quart==0.19.3,redis==5.0.1,requests==2.31.0,ruff==0.1.3,setuptools==68.2.2,sniffio==1.3.0,taskgroup==0.0.0a4,tomli==2.0.1,tomli_w==1.0.0,tosholi==0.1.0,tox==4.11.3,types-pyOpenSSL==23.3.0.0,types-redis==4.6.0.8,types-requests==2.31.0.10,typing_extensions==4.8.0,urllib3==2.0.7,uvicorn==0.23.2,uvloop==0.19.0,virtualenv==20.24.6,watchfiles==0.21.0,websockets==12.0,Werkzeug==3.0.1,wheel==0.41.2,wsproto==1.2.0
2023-10-30T19:12:19.5021220Z py310: commands[0]> coverage erase
2023-10-30T19:12:19.6945100Z py310: commands[1]> coverage run -m pytest
2023-10-30T19:12:23.5690730Z ============================= test session starts ==============================
2023-10-30T19:12:23.5692850Z platform darwin -- Python 3.10.13, pytest-7.4.3, pluggy-1.3.0
2023-10-30T19:12:23.5694640Z cachedir: .tox/py310/.pytest_cache
2023-10-30T19:12:23.5697430Z rootdir: /Users/runner/work/proxystore/proxystore
2023-10-30T19:12:23.5698780Z configfile: pyproject.toml
2023-10-30T19:12:23.5700210Z plugins: anyio-4.0.0, timeout-2.2.0, cov-4.1.0, asyncio-0.21.1
2023-10-30T19:12:23.5701560Z asyncio: mode=strict
2023-10-30T19:12:23.5704000Z collected 420 items
2023-10-30T19:12:23.5704510Z 
2023-10-30T19:12:23.5761660Z tests/factory_test.py ..                                                 [  0%]
2023-10-30T19:12:23.5851470Z tests/proxy_test.py ....                                                 [  1%]
2023-10-30T19:12:23.5877860Z tests/serialization_test.py .                                            [  1%]
2023-10-30T19:12:25.3796480Z tests/connectors/connector_test.py ..............................        [  8%]
2023-10-30T19:12:25.3838740Z tests/connectors/dim_test.py .                                           [  9%]
2023-10-30T19:12:25.4679560Z tests/connectors/endpoint_test.py ......                                 [ 10%]
2023-10-30T19:12:25.4832400Z tests/connectors/file_test.py ....                                       [ 11%]
2023-10-30T19:12:25.5180550Z tests/connectors/globus_test.py ............                             [ 14%]
2023-10-30T19:12:25.5202660Z tests/connectors/local_test.py .                                         [ 14%]
2023-10-30T19:12:25.5564020Z tests/connectors/multi_test.py .................                         [ 18%]
2023-10-30T19:12:25.5626090Z tests/connectors/redis_test.py ...                                       [ 19%]
2023-10-30T19:12:25.7353680Z tests/endpoint/cli_test.py ............                                  [ 22%]
2023-10-30T19:12:25.8050260Z tests/endpoint/client_test.py ...                                        [ 22%]
2023-10-30T19:12:26.0185520Z tests/endpoint/commands_test.py ........................                 [ 28%]
2023-10-30T19:12:26.1164450Z tests/endpoint/config_test.py .............................              [ 35%]
2023-10-30T19:12:26.6824020Z tests/endpoint/endpoint_peering_test.py ...........                      [ 38%]
2023-10-30T19:12:26.7186460Z tests/endpoint/endpoint_solo_test.py ......                              [ 39%]
2023-10-30T19:12:28.6667680Z tests/endpoint/serve_test.py ................                            [ 43%]
2023-10-30T19:12:28.7175730Z tests/endpoint/storage_test.py .....                                     [ 44%]
2023-10-30T19:12:28.7685280Z tests/globus/cli_test.py ....                                            [ 45%]
2023-10-30T19:12:28.7948920Z tests/globus/client_test.py .....                                        [ 46%]
2023-10-30T19:12:28.8811850Z tests/globus/manager_test.py .....                                       [ 47%]
2023-10-30T19:12:28.9083990Z tests/globus/scopes_test.py .....                                        [ 49%]
2023-10-30T19:12:28.9288710Z tests/globus/storage_test.py ..                                          [ 49%]
2023-10-30T19:12:28.9934840Z tests/globus/transfer_test.py ....                                       [ 50%]
2023-10-30T19:12:42.1383130Z tests/integration/endpoints_test.py ...                                  [ 51%]
2023-10-30T19:12:42.2384620Z tests/p2p/chunks_test.py ......                                          [ 52%]
2023-10-30T19:12:43.5248650Z tests/p2p/connection_test.py ......                                      [ 54%]
2023-10-30T19:12:45.6605590Z tests/p2p/manager_test.py .....F.......                                  [ 57%]
2023-10-30T19:12:45.6942350Z tests/p2p/relay/authenticate_test.py ...........                         [ 59%]
2023-10-30T19:12:46.0257120Z tests/p2p/relay/client_test.py ................                          [ 63%]
2023-10-30T19:12:46.0467590Z tests/p2p/relay/config_test.py ....                                      [ 64%]
2023-10-30T19:12:46.0682760Z tests/p2p/relay/manager_test.py ...                                      [ 65%]
2023-10-30T19:12:46.1135680Z tests/p2p/relay/messages_test.py ..................                      [ 69%]
2023-10-30T19:12:47.9520900Z tests/p2p/relay/run_test.py ......                                       [ 70%]
2023-10-30T19:12:48.1578090Z tests/p2p/relay/server_test.py ..................                        [ 75%]
2023-10-30T19:12:48.1697690Z tests/store/cache_test.py ..                                             [ 75%]
2023-10-30T19:12:48.1859360Z tests/store/init_test.py ....                                            [ 76%]
2023-10-30T19:12:48.2050130Z tests/store/metrics_test.py ........                                     [ 78%]
2023-10-30T19:12:48.2553210Z tests/store/store_basics_test.py ..........                              [ 80%]
2023-10-30T19:12:48.2746570Z tests/store/store_metrics_test.py ..                                     [ 81%]
2023-10-30T19:12:48.3406960Z tests/store/store_proxy_test.py .....................                    [ 86%]
2023-10-30T19:12:48.3486180Z tests/store/utils_test.py ...                                            [ 87%]
2023-10-30T19:12:48.3533610Z tests/utils/counter_test.py ..                                           [ 87%]
2023-10-30T19:12:48.4443710Z tests/utils/data_test.py .....................................           [ 96%]
2023-10-30T19:12:48.4555470Z tests/utils/environment_test.py ....                                     [ 97%]
2023-10-30T19:12:48.4695580Z tests/utils/imports_test.py ......                                       [ 98%]
2023-10-30T19:12:48.4843270Z tests/utils/tasks_test.py ..                                             [ 99%]
2023-10-30T19:12:48.9129140Z tests/utils/timer_test.py ...                                            [100%]
2023-10-30T19:12:48.9130570Z 
2023-10-30T19:12:48.9132550Z =================================== FAILURES ===================================
2023-10-30T19:12:48.9134440Z ____________________ test_p2p_connection_error_from_server _____________________
2023-10-30T19:12:48.9135950Z 
2023-10-30T19:12:48.9136460Z self = <_SelectorDatagramTransport closed fd=39>
2023-10-30T19:12:48.9144890Z data = b'\x00\x01\x00H!\x12\xa4BQ\x99\xf9\x7f\xca\xc9+\xf8\n\xd0\xd5\x1e\x00\x06\x00\tTbvQ:s0aE\x00\x00\x00\x00$\x00\x04n\xff...\xc6\x00%\x00\x00\x00\x08\x00\x14\x12\x11\xb5\x1dAM\xa2\x8eB\xde\xe8m\xa9\xef\xabl\x19]_\xaa\x80(\x00\x04V\x0e\xc2\xc8'
2023-10-30T19:12:48.9150260Z addr = ('13.105.117.62', 45837)
2023-10-30T19:12:48.9151140Z 
2023-10-30T19:12:48.9151820Z     def sendto(self, data, addr=None):
2023-10-30T19:12:48.9154160Z         if not isinstance(data, (bytes, bytearray, memoryview)):
2023-10-30T19:12:48.9157020Z             raise TypeError(f'data argument must be a bytes-like object, '
2023-10-30T19:12:48.9158330Z                             f'not {type(data).__name__!r}')
2023-10-30T19:12:48.9159100Z         if not data:
2023-10-30T19:12:48.9160130Z             return
2023-10-30T19:12:48.9160630Z     
2023-10-30T19:12:48.9161230Z         if self._address:
2023-10-30T19:12:48.9161900Z             if addr not in (None, self._address):
2023-10-30T19:12:48.9163410Z                 raise ValueError(
2023-10-30T19:12:48.9164880Z                     f'Invalid address: must be None or {self._address}')
2023-10-30T19:12:48.9165810Z             addr = self._address
2023-10-30T19:12:48.9166790Z     
2023-10-30T19:12:48.9167900Z         if self._conn_lost and self._address:
2023-10-30T19:12:48.9169100Z             if self._conn_lost >= constants.LOG_THRESHOLD_FOR_CONNLOST_WRITES:
2023-10-30T19:12:48.9170770Z                 logger.warning('socket.send() raised exception.')
2023-10-30T19:12:48.9172450Z             self._conn_lost += 1
2023-10-30T19:12:48.9173630Z             return
2023-10-30T19:12:48.9174300Z     
2023-10-30T19:12:48.9175310Z         if not self._buffer:
2023-10-30T19:12:48.9176130Z             # Attempt to send it right away first.
2023-10-30T19:12:48.9177060Z             try:
2023-10-30T19:12:48.9177770Z                 if self._extra['peername']:
2023-10-30T19:12:48.9178690Z                     self._sock.send(data)
2023-10-30T19:12:48.9179400Z                 else:
2023-10-30T19:12:48.9180010Z >                   self._sock.sendto(data, addr)
2023-10-30T19:12:48.9181110Z E                   AttributeError: 'NoneType' object has no attribute 'sendto'
2023-10-30T19:12:48.9181920Z 
2023-10-30T19:12:48.9183230Z /usr/local/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/selector_events.py:1062: AttributeError
2023-10-30T19:12:48.9185560Z 
2023-10-30T19:12:48.9186060Z During handling of the above exception, another exception occurred:
2023-10-30T19:12:48.9186900Z 
2023-10-30T19:12:48.9189750Z relay_server = RelayServerInfo(relay_server=<proxystore.p2p.relay.server.RelayServer object at 0x10a06a830>, websocket_server=<websockets.legacy.server.WebSocketServer object at 0x10a06a860>, host='localhost', port=49319, address='ws://localhost:49319')
2023-10-30T19:12:48.9192730Z 
2023-10-30T19:12:48.9192980Z     @pytest.mark.asyncio()
2023-10-30T19:12:48.9194060Z     async def test_p2p_connection_error_from_server(relay_server) -> None:
2023-10-30T19:12:48.9195280Z         # Record current tasks so we know which not to clean up
2023-10-30T19:12:48.9196430Z         task_names = {task.get_name() for task in asyncio.all_tasks()}
2023-10-30T19:12:48.9197320Z     
2023-10-30T19:12:48.9198470Z         async with PeerManager(
2023-10-30T19:12:48.9199300Z             RelayClient(relay_server.address),
2023-10-30T19:12:48.9200110Z         ) as manager1, PeerManager(
2023-10-30T19:12:48.9200900Z             RelayClient(relay_server.address),
2023-10-30T19:12:48.9201760Z         ) as manager2:
2023-10-30T19:12:48.9202660Z             # Mock manager 1 to receive error peer connection from relay server
2023-10-30T19:12:48.9203950Z             mock_recv = async_mock_once(
2023-10-30T19:12:48.9204730Z                 manager1._relay_client.recv,
2023-10-30T19:12:48.9205530Z                 PeerConnectionRequest(
2023-10-30T19:12:48.9206420Z                     source_uuid=manager2.uuid,
2023-10-30T19:12:48.9207220Z                     source_name=manager2.name,
2023-10-30T19:12:48.9208020Z                     peer_uuid=manager1.uuid,
2023-10-30T19:12:48.9209100Z                     description_type='offer',
2023-10-30T19:12:48.9209930Z                     description='',
2023-10-30T19:12:48.9210660Z                     error='test error',
2023-10-30T19:12:48.9211500Z                 ),
2023-10-30T19:12:48.9212000Z             )
2023-10-30T19:12:48.9212960Z             manager1._relay_client.recv = mock_recv  # type: ignore
2023-10-30T19:12:48.9213860Z     
2023-10-30T19:12:48.9214510Z             connection1 = await manager1.get_connection(manager2.uuid)
2023-10-30T19:12:48.9215410Z     
2023-10-30T19:12:48.9215950Z             while not mock_recv.await_count > 1:
2023-10-30T19:12:48.9216770Z                 await asyncio.sleep(0.01)
2023-10-30T19:12:48.9217440Z     
2023-10-30T19:12:48.9218580Z             with pytest.raises(PeerConnectionError, match='test error'):
2023-10-30T19:12:48.9219570Z                 await connection1.ready()
2023-10-30T19:12:48.9220830Z     
2023-10-30T19:12:48.9221890Z         # Clean up tasks that were left pending because we raised an exception
2023-10-30T19:12:48.9223000Z         for task in asyncio.all_tasks():
2023-10-30T19:12:48.9223820Z             if task.get_name() not in task_names:
2023-10-30T19:12:48.9224570Z                 task.cancel()
2023-10-30T19:12:48.9225160Z                 try:
2023-10-30T19:12:48.9226090Z >                   await task
2023-10-30T19:12:48.9226610Z 
2023-10-30T19:12:48.9226890Z tests/p2p/manager_test.py:114: 
2023-10-30T19:12:48.9228970Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2023-10-30T19:12:48.9230470Z .tox/py310/lib/python3.10/site-packages/aioice/ice.py:797: in check_start
2023-10-30T19:12:48.9231680Z     response, addr = await pair.protocol.request(
2023-10-30T19:12:48.9233050Z .tox/py310/lib/python3.10/site-packages/aioice/ice.py:254: in request
2023-10-30T19:12:48.9234150Z     return await transaction.run()
2023-10-30T19:12:48.9235240Z .tox/py310/lib/python3.10/site-packages/aioice/stun.py:301: in run
2023-10-30T19:12:48.9236390Z     self.__retry()
2023-10-30T19:12:48.9237360Z .tox/py310/lib/python3.10/site-packages/aioice/stun.py:312: in __retry
2023-10-30T19:12:48.9238570Z     self.__protocol.send_stun(self.__request, self.__addr)
2023-10-30T19:12:48.9240050Z .tox/py310/lib/python3.10/site-packages/aioice/ice.py:266: in send_stun
2023-10-30T19:12:48.9241740Z     self.transport.sendto(bytes(message), addr)
2023-10-30T19:12:48.9243610Z /usr/local/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/selector_events.py:1072: in sendto
2023-10-30T19:12:48.9245530Z     self._fatal_error(
2023-10-30T19:12:48.9247690Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2023-10-30T19:12:48.9248630Z 
2023-10-30T19:12:48.9248980Z self = <_SelectorDatagramTransport closed fd=39>
2023-10-30T19:12:48.9250580Z exc = AttributeError("'NoneType' object has no attribute 'sendto'")
2023-10-30T19:12:48.9251820Z message = 'Fatal write error on datagram transport'
2023-10-30T19:12:48.9252460Z 
2023-10-30T19:12:48.9253010Z     def _fatal_error(self, exc, message='Fatal error on transport'):
2023-10-30T19:12:48.9254110Z         # Should be called from exception handler only.
2023-10-30T19:12:48.9254970Z         if isinstance(exc, OSError):
2023-10-30T19:12:48.9255670Z             if self._loop.get_debug():
2023-10-30T19:12:48.9256530Z                 logger.debug("%r: %s", self, message, exc_info=True)
2023-10-30T19:12:48.9257360Z         else:
2023-10-30T19:12:48.9258040Z >           self._loop.call_exception_handler({
2023-10-30T19:12:48.9258970Z                 'message': message,
2023-10-30T19:12:48.9259730Z                 'exception': exc,
2023-10-30T19:12:48.9260470Z                 'transport': self,
2023-10-30T19:12:48.9261250Z                 'protocol': self._protocol,
2023-10-30T19:12:48.9261930Z             })
2023-10-30T19:12:48.9263010Z E           AttributeError: 'NoneType' object has no attribute 'call_exception_handler'
2023-10-30T19:12:48.9264330Z 
2023-10-30T19:12:48.9265670Z /usr/local/Cellar/python@3.10/3.10.13/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/selector_events.py:719: AttributeError
2023-10-30T19:12:48.9268130Z ------------------------------ Captured log call -------------------------------
2023-10-30T19:12:48.9271080Z [2023-10-30 19:12:43.904] INFO  (proxystore.p2p.relay.server) :: Unregistering client e30d84a8-9ad4-4fb8-9726-efe5a371e94b (Mac-1698692463362.local) for ok reason
2023-10-30T19:12:48.9273440Z [2023-10-30 19:12:43.904] INFO  (websockets.server) :: connection closed
2023-10-30T19:12:48.9274980Z [2023-10-30 19:12:43.910] INFO  (websockets.server) :: connection open
2023-10-30T19:12:48.9278100Z [2023-10-30 19:12:43.912] INFO  (proxystore.p2p.relay.server) :: Registered client: Client(name=Mac-1698692463362.local, uuid=c9648a29-7485-4307-b8ce-34b8b042d6e3, user=NullUser(), address=('::1', 49846, 0, 0), created=2023-10-30 19:12:43 UTC)
2023-10-30T19:12:48.9282710Z [2023-10-30 19:12:43.914] INFO  (proxystore.p2p.relay.client) :: Established client connection to relay server at ws://localhost:49319 with client uuid=c9648a29-7485-4307-b8ce-34b8b042d6e3 and name=Mac-1698692463362.local
2023-10-30T19:12:48.9286370Z [2023-10-30 19:12:43.915] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(c9648a29)]: listening for messages from relay server
2023-10-30T19:12:48.9288480Z [2023-10-30 19:12:43.918] INFO  (websockets.server) :: connection open
2023-10-30T19:12:48.9292240Z [2023-10-30 19:12:43.920] INFO  (proxystore.p2p.relay.server) :: Registered client: Client(name=Mac-1698692463362.local, uuid=697f3039-47e7-4642-a4f9-2ca8b07c338e, user=NullUser(), address=('::1', 49847, 0, 0), created=2023-10-30 19:12:43 UTC)
2023-10-30T19:12:48.9296870Z [2023-10-30 19:12:43.923] INFO  (proxystore.p2p.relay.client) :: Established client connection to relay server at ws://localhost:49319 with client uuid=697f3039-47e7-4642-a4f9-2ca8b07c338e and name=Mac-1698692463362.local
2023-10-30T19:12:48.9300750Z [2023-10-30 19:12:43.928] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(c9648a29)]: opening peer connection with 697f3039-47e7-4642-a4f9-2ca8b07c338e
2023-10-30T19:12:48.9303880Z [2023-10-30 19:12:43.931] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(697f3039)]: listening for messages from relay server
2023-10-30T19:12:48.9307740Z [2023-10-30 19:12:44.000] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(c9648a29) > pending]: sending offer to 697f3039-47e7-4642-a4f9-2ca8b07c338e
2023-10-30T19:12:48.9311840Z [2023-10-30 19:12:44.003] INFO  (proxystore.p2p.relay.server) :: Transmitting message from c9648a29-7485-4307-b8ce-34b8b042d6e3 (Mac-1698692463362.local) to 697f3039-47e7-4642-a4f9-2ca8b07c338e (Mac-1698692463362.local)
2023-10-30T19:12:48.9317160Z [2023-10-30 19:12:44.007] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(697f3039) > pending]: received offer from c9648a29-7485-4307-b8ce-34b8b042d6e3 (Mac-1698692463362.local)
2023-10-30T19:12:48.9321220Z [2023-10-30 19:12:44.082] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(697f3039) > Mac-1698692463362.local(c9648a29)]: sending answer to c9648a29-7485-4307-b8ce-34b8b042d6e3
2023-10-30T19:12:48.9324690Z [2023-10-30 19:12:44.083] INFO  (aioice.ice) :: Connection(16) Check CandidatePair(('192.168.64.24', 61176) -> ('192.168.64.24', 59129)) State.FROZEN -> State.WAITING
2023-10-30T19:12:48.9327710Z [2023-10-30 19:12:44.083] INFO  (aioice.ice) :: Connection(16) Check CandidatePair(('192.168.64.24', 61176) -> ('13.105.117.62', 47231)) State.FROZEN -> State.WAITING
2023-10-30T19:12:48.9330850Z [2023-10-30 19:12:44.084] INFO  (aioice.ice) :: Connection(16) Check CandidatePair(('192.168.64.24', 61176) -> ('192.168.64.24', 59129)) State.WAITING -> State.IN_PROGRESS
2023-10-30T19:12:48.9335100Z [2023-10-30 19:12:44.085] INFO  (proxystore.p2p.relay.server) :: Transmitting message from 697f3039-47e7-4642-a4f9-2ca8b07c338e (Mac-1698692463362.local) to c9648a29-7485-4307-b8ce-34b8b042d6e3 (Mac-1698692463362.local)
2023-10-30T19:12:48.9338900Z [2023-10-30 19:12:44.086] INFO  (aioice.ice) :: Connection(16) Check CandidatePair(('192.168.64.24', 61176) -> ('192.168.64.24', 59129)) State.IN_PROGRESS -> State.SUCCEEDED
2023-10-30T19:12:48.9342780Z [2023-10-30 19:12:44.088] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(c9648a29) > pending]: received answer from 697f3039-47e7-4642-a4f9-2ca8b07c338e (Mac-1698692463362.local)
2023-10-30T19:12:48.9346490Z [2023-10-30 19:12:44.093] INFO  (aioice.ice) :: Connection(15) Check CandidatePair(('192.168.64.24', 59129) -> ('192.168.64.24', 61176)) State.FROZEN -> State.WAITING
2023-10-30T19:12:48.9349850Z [2023-10-30 19:12:44.093] INFO  (aioice.ice) :: Connection(15) Check CandidatePair(('192.168.64.24', 59129) -> ('13.105.117.62', 45837)) State.FROZEN -> State.WAITING
2023-10-30T19:12:48.9353010Z [2023-10-30 19:12:44.094] INFO  (aioice.ice) :: Connection(15) Check CandidatePair(('192.168.64.24', 59129) -> ('192.168.64.24', 61176)) State.WAITING -> State.IN_PROGRESS
2023-10-30T19:12:48.9355900Z [2023-10-30 19:12:44.094] ERROR (proxystore.p2p.manager) :: Received error message from relay server: test error
2023-10-30T19:12:48.9359320Z [2023-10-30 19:12:44.094] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(c9648a29)] Closing connection between peers: c9648a29-7485-4307-b8ce-34b8b042d6e3, 697f3039-47e7-4642-a4f9-2ca8b07c338e
2023-10-30T19:12:48.9363720Z [2023-10-30 19:12:44.094] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(c9648a29) > Mac-1698692463362.local(697f3039)]: closing connection
2023-10-30T19:12:48.9367540Z [2023-10-30 19:12:44.095] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(c9648a29) > Mac-1698692463362.local(697f3039)]: connection entered closed state, invoking callback
2023-10-30T19:12:48.9371350Z [2023-10-30 19:12:44.096] INFO  (aioice.ice) :: Connection(16) Check CandidatePair(('192.168.64.24', 61176) -> ('13.105.117.62', 47231)) State.WAITING -> State.FAILED
2023-10-30T19:12:48.9373670Z [2023-10-30 19:12:44.097] INFO  (aioice.ice) :: Connection(16) ICE completed
2023-10-30T19:12:48.9376270Z [2023-10-30 19:12:44.099] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(697f3039) > Mac-1698692463362.local(c9648a29)]: closing connection
2023-10-30T19:12:48.9380510Z [2023-10-30 19:12:44.100] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(697f3039) > Mac-1698692463362.local(c9648a29)]: connection entered closed state, invoking callback
2023-10-30T19:12:48.9384720Z [2023-10-30 19:12:44.103] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(697f3039)] Closing connection between peers: c9648a29-7485-4307-b8ce-34b8b042d6e3, 697f3039-47e7-4642-a4f9-2ca8b07c338e
2023-10-30T19:12:48.9388650Z [2023-10-30 19:12:44.103] INFO  (proxystore.p2p.connection) :: PeerConnection[Mac-1698692463362.local(697f3039) > Mac-1698692463362.local(c9648a29)]: closing connection
2023-10-30T19:12:48.9391590Z [2023-10-30 19:12:44.109] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(697f3039)]: peer manager closed
2023-10-30T19:12:48.9415250Z [2023-10-30 19:12:44.110] INFO  (proxystore.p2p.relay.server) :: Unregistering client 697f3039-47e7-4642-a4f9-2ca8b07c338e (Mac-1698692463362.local) for ok reason
2023-10-30T19:12:48.9417560Z [2023-10-30 19:12:44.111] INFO  (websockets.server) :: connection closed
2023-10-30T19:12:48.9419560Z [2023-10-30 19:12:44.114] INFO  (proxystore.p2p.manager) :: PeerManager[Mac-1698692463362.local(c9648a29)]: peer manager closed
2023-10-30T19:12:48.9422510Z [2023-10-30 19:12:44.115] INFO  (aioice.ice) :: Connection(15) Check CandidatePair(('192.168.64.24', 59129) -> ('13.105.117.62', 45837)) State.WAITING -> State.IN_PROGRESS
2023-10-30T19:12:48.9424780Z [2023-10-30 19:12:44.116] INFO  (websockets.server) :: connection closed
2023-10-30T19:12:48.9425860Z =============================== warnings summary ===============================
2023-10-30T19:12:48.9427590Z .tox/py310/lib/python3.10/site-packages/google_crc32c/__config__.py:18
2023-10-30T19:12:48.9430960Z   /Users/runner/work/proxystore/proxystore/.tox/py310/lib/python3.10/site-packages/google_crc32c/__config__.py:18: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
2023-10-30T19:12:48.9433750Z     import pkg_resources
2023-10-30T19:12:48.9434100Z 
2023-10-30T19:12:48.9434700Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2023-10-30T19:12:48.9435820Z =========================== short test summary info ============================
2023-10-30T19:12:48.9437170Z FAILED tests/p2p/manager_test.py::test_p2p_connection_error_from_server - Att...
2023-10-30T19:12:48.9438470Z ================== 1 failed, 419 passed, 1 warning in 27.05s ===================
2023-10-30T19:12:49.2432670Z py310: exit 1 (29.51 seconds) /Users/runner/work/proxystore/proxystore> coverage run -m pytest pid=3837
2023-10-30T19:12:49.2452220Z .pkg: _exit> python /Users/runner/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/pyproject_api/_backend.py True setuptools.build_meta
2023-10-30T19:12:49.3010530Z   py310: FAIL code 1 (74.79=setup[45.08]+cmd[0.20,29.51] seconds)
2023-10-30T19:12:49.3011620Z   evaluation failed :( (74.99 seconds)
2023-10-30T19:12:49.3533720Z ##[error]Process completed with exit code 1.
2023-10-30T19:12:49.3719300Z Post job cleanup.
2023-10-30T19:12:49.4991430Z [command]/usr/local/bin/git version
2023-10-30T19:12:49.5076490Z git version 2.42.0
2023-10-30T19:12:49.5128600Z Copying '/Users/runner/.gitconfig' to '/Users/runner/work/_temp/37598208-7b0b-4691-b11a-7e127f258494/.gitconfig'
2023-10-30T19:12:49.5152120Z Temporarily overriding HOME='/Users/runner/work/_temp/37598208-7b0b-4691-b11a-7e127f258494' before making global git config changes
2023-10-30T19:12:49.5154500Z Adding repository directory to the temporary git global config as a safe directory
2023-10-30T19:12:49.5157580Z [command]/usr/local/bin/git config --global --add safe.directory /Users/runner/work/proxystore/proxystore
2023-10-30T19:12:49.5250680Z [command]/usr/local/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-10-30T19:12:49.5332910Z [command]/usr/local/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-10-30T19:12:49.6556820Z [command]/usr/local/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-10-30T19:12:49.6559380Z http.https://github.com/.extraheader
2023-10-30T19:12:49.6584750Z [command]/usr/local/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-10-30T19:12:49.6670600Z [command]/usr/local/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-10-30T19:12:49.8055260Z Cleaning up orphan processes
```

### How did you install ProxyStore?

```console
N/A
```


### ProxyStore Version

HEAD

### Python Version

ALL

### OS and Platform

MacOS

--- a/tests/p2p/manager_test.py
+++ b/tests/p2p/manager_test.py
@@ -112,7 +112,9 @@ async def test_p2p_connection_error_from_server(relay_server) -> None:
             task.cancel()
             try:
                 await task
-            except asyncio.CancelledError:
+            # For note on AttributeError catching:
+            # https://github.com/proxystore/proxystore/issues/405
+            except (asyncio.CancelledError, AttributeError):
                 pass
 
 

